src/app/model/ImageDataValidators.ts

```diff
@@ -0,0 +1,56 @@
+import { ValidatorFn } from '@angular/forms';
+import dayjs from 'dayjs';
+
+export class ImageDataValidators {
+    static readonly LONG_MAX_VAL = 2_147_483_647;
+    static readonly SHORT_MAX_VAL = 65_536;
+    static readonly ASCII_MAX_CHARS = 128;
+    static readonly ZERO = 0;
+
+    static date(): ValidatorFn {
+        return (control) => {
+            if (dayjs(control.value).isValid()) {
+                return null;
+            } else {
+                return { message: 'Date value is invalid' };
+            }
+        };
+    }
+
+    static min(min: number): ValidatorFn {
```

Comment:
Yes, I am aware of it. Unfortunately angular validators returns different key in error map depending on validator. This would cause and issue for me when displaying error using generic table. I will try to simplify this code.

Reply:
For min/max value and max length validator I added usage of defined Angular validators but wrapped them to return errors with expected in form object inside.